packages/gatsby-transformer-remark/src/extend-node-type.js

```diff
@@ -242,13 +250,29 @@ module.exports = (
       }
     }
 
-    async function getTableOfContents(markdownNode, pathToSlugField) {
+    async function getTableOfContents(
+      markdownNode,
+      pathToSlugField,
+      maxDepth,
+      heading,
+      tight
+    ) {
       const cachedToc = await cache.get(tableOfContentsCacheKey(markdownNode))
```

Comment:
cache key would need to contain options, otherwise if you query same markdown node with different toc options it will return same output even if options are different

Reply:
fixed in new commit